small-talk/app/homepage/page.js

```diff
@@ -1,14 +1,156 @@
+    backgroundColor: '#007bff',
+    color: '#fff',
+    cursor: 'pointer',
+    transition: 'transform 0.3s ease', // Transition for the sliding effect
+    margin: '5px 0', // Margin between buttons
+    textAlign: 'left', // Align text to the left
+    fontSize: '18px', // Increase font size if needed
+    height: '10%', // Each button will take up 10% of the sidebar height
+    flexGrow: 1, // Allow buttons to grow and fill the container
+
+};
+
+const hoverEffect = {
+    transform: 'translateX(10px)', // Slide to the right on hover
+};
+
+
+const SidebarStyle = {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start', // Align buttons to the left side
+    position: 'fixed', // Make sidebar fixed to the side of the screen
+    left: 0,
+    top: 0,
+    width: '250px', // Width of the sidebar
+    height: '100%', // Sidebar spans the full height of the viewport
+    backgroundColor: '#333', // Light background for the sidebar
+    padding: '10px',
+    boxSizing: 'border-box',
+    justifyContent: 'space-between', // Distribute buttons evenly
+
+};
+
+
+const HeaderStyle = {
+    color: '#FFFFFF', // A bright color for better visibility on a black background
+    fontSize: '24px', // Keeping the font size large for better readability
+    position: 'absolute', // Positioning it absolutely will take it out of the normal document flow
+    top: '50%', // Position it halfway down the parent container
+    left: '50%', // Position it halfway across the parent container
+    transform: 'translate(-50%, -50%)', // Offset the header by half its width and height for centering
+    textAlign: 'center', // Center-align the text
+    zIndex: '10', // Ensure it's above other elements (if necessary)
+};
+
+
+
+const HoverButton = ({ style, onClick, children }) => {
+    const [isHovered, setIsHovered] = useState(false);
+
+    const hoverStyle = {
+        ...style,
+        ...(isHovered ? hoverEffect : null),
+    };
+
+    return (
+        <button
+            style={hoverStyle}
+            onClick={onClick}
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+        >
+            {children}
+        </button>
+    );
+};
+
+
+const handleGame = () => {
+
+    window.location.href = "/game"; // Redirect to the game
+};
+
+const handleChat = () => {
+
+    window.location.href = "/chat"; // Redirect to the chat
+};
+
+const handleFood = () => {
+
+    window.location.href = "/food"; // Redirect to the food
+};
+
+const handleMovies = () => {
+
+    window.location.href = "/movie"; // Redirect to the movies
+};
+
+const handleSettings = () => {
+
+    window.location.href = "/setting"; // Redirect to the setting
+};
+
+const handleProfile = () => {
+
+    window.location.href = "/profile"; // Redirect to the profile
+};
+
+const handleSignOut = () => {
+
+    window.location.href = "/"; // Redirect to the welcome screen
```

Comment:
A lot of the functions are pretty similar and could be refactored into a single function that takes the destination as an argument, which would reduce redundancy. May not have to happen on this PR but should keep in mind for future code tidiness.